operational_analysis/methods/plant_analysis.py

```diff
@@ -854,8 +860,33 @@ def sample_long_term_reanalysis(self, n, r):
             wd_aggregate = np.pi-np.arctan2(-long_term_temp[r + '_u_ms'],long_term_temp[r + '_v_ms']) # Calculate wind direction
             long_term_reg_inputs = pd.concat([long_term_reg_inputs, np.sin(wd_aggregate), np.cos(wd_aggregate)], axis=1)
         
+        # Store result in dictionary
+        self.long_term_sampling[(r, n)] = long_term_reg_inputs
+        self.long_term_sampling_sd[(r, n)] = ws_aggregate_sd
+        
         # Return result            
-        return long_term_reg_inputs
+        return [long_term_reg_inputs,ws_aggregate_sd]
+
+    @logged_method_call
+    def apply_IAV(self, long_term_reg_inputs, ws_aggregate_sd_IAV, r):
+        """
+        This function calculates IAV for the long-term wind speed data
+        
+        Args:
+           long_term_reg_inputs(:obj:`pandas.DataFrame`): Long-term regression inputs
+           ws_aggregate_sd(:obj:`pandas.DataFrame`): Long-term wind speed stdev, from which IAV will be derived
+           r(:obj:`string`): The reanalysis product used for the current Monte Carlo iteration
+        Returns:
+           :obj:`pandas.DataFrame`: the IAV-sampled monthly/daily regression inputs
+        """
+    
+        if self.time_resolution == 'D':
+            ws_aggregate_sd_IAV = np.repeat(ws_aggregate_sd_IAV, np.ceil(self.num_days_lt).astype(int))
+        ws_aggregate_sd_IAV.reset_index(drop=True,inplace=True) # TODO: This rectifies a 0-indexing issue. Needs refactoring.
+        iav = np.random.normal(1, ws_aggregate_sd_IAV/long_term_reg_inputs[r])        
+        long_term_reg_inputs_IAV = long_term_reg_inputs.copy()
```

Comment:
I agree this copy() is not nice to see. However, without it I noticed some weird behaviors where the wind speed variable after IAV is applied would overwrite the corresponding original dictionary of the memoized approach in sample_long_term_reanalysis. Basically, self.long_term_sampling[(r, n)] would be modified. This is also why I have added the copy() commands in your later comment here... If you know how to avoid this, please let me know!

Reply:
I would prefer to see this copy and the copies on lines 837-838 deleted and then replace lines 839 and 868 with the following:

```
return [long_term_reg_inputs.copy(),ws_aggregate_sd.copy()]
```

This way it is clear when the copy is happening (every time you call sample_long_term_reanalysis)

Reply:
Looks great to me! Thank you!